src/main/java/de/tum/in/www1/artemis/service/competency/KnowledgeAreaService.java

```diff
@@ -0,0 +1,54 @@
+package de.tum.in.www1.artemis.service.competency;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import javax.ws.rs.BadRequestException;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.competency.KnowledgeArea;
+import de.tum.in.www1.artemis.repository.competency.KnowledgeAreaRepository;
+
+/**
+ * Service for managing {@link KnowledgeArea} entities.
+ */
+@Profile(PROFILE_CORE)
+@Service
+public class KnowledgeAreaService {
+
+    private final KnowledgeAreaRepository knowledgeAreaRepository;
+
+    public KnowledgeAreaService(KnowledgeAreaRepository knowledgeAreaRepository) {
+        this.knowledgeAreaRepository = knowledgeAreaRepository;
+    }
+
+    public KnowledgeArea createKnowledgeArea(KnowledgeArea knowledgeArea) {
+        knowledgeAreaIsValidOrElseThrow(knowledgeArea);
+
+        // fetch the parent from the database if it exists
+        KnowledgeArea parent = knowledgeArea.getParent();
+        if (parent != null) {
+            parent = knowledgeAreaRepository.findByIdElseThrow(parent.getId());
+        }
+
+        var knowledgeAreaToCreate = new KnowledgeArea(knowledgeArea.getTitle(), knowledgeArea.getDescription());
+        knowledgeAreaToCreate.setParent(parent);
+        var createdKnowledgeArea = knowledgeAreaRepository.save(knowledgeAreaToCreate);
+        // set the parent as it is lazily loaded
+        createdKnowledgeArea.setParent(parent);
+
+        return createdKnowledgeArea;
+    }
+
+    /**
+     * Verifies that a knowledge area that should be created is valid or throws a BadRequestException
+     *
+     * @param knowledgeArea the knowledge area to verify
+     */
+    public void knowledgeAreaIsValidOrElseThrow(KnowledgeArea knowledgeArea) throws BadRequestException {
```

Comment:
Should this be public? Any use of this method outside of this service would indicate that the logic should probably be part of it.